sdk/invoke.js

```diff
@@ -33,24 +34,24 @@ const invoke = async function invoke(accountId, funcName, apiKey, funcData) {
     logger.debug('Invoke response', { statusCode, headers, body });
     return res;
   } catch (err) {
-    let parsedError;
-    try {
-      parsedError = JSON.parse(err.error);
-    } catch (nestedErr) {
-      logger.debug('Failed to parse JSON response', { nestedErr });
-    }
-
-    if (parsedError) {
-      const errorOrMsg = parsedError.error || parsedError.message;
-      if (parsedError.stack && errorOrMsg) {
-        throw new Error(`${errorOrMsg}\n${parsedError.stack}`);
+    const fallbackError = err.error || err;
+    if (err instanceof StatusCodeError) {
+      const requestId = err.response.headers['x-binaris-request-id'];
+      let parsedError;
+      try {
+        parsedError = JSON.parse(err.error);
+      } catch (parseError) {
+        parsedError = { error: fallbackError };
       }
-      if (parsedError.message && parsedError.request_id) {
-        throw new Error(`${parsedError.message}\nrequest_id: ${parsedError.request_id}`);
+      const fullError = [`RequestId: ${requestId}`, parsedError.error || parsedError.message];
+
+      if (parsedError.stack) {
+        fullError.push(parsedError.stack);
       }
+      throw new Error(fullError.join('\n'));
     }
 
-    throw new Error(err.error || err);
+    throw new Error(fallbackError);
```

Comment:
if err is not instanceof StatusCodeError how can err.error be meaningful? shouldn't this line be `throw err`?